servlet/src/main/java/com/linecorp/armeria/server/servlet/UrlMapper.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.servlet;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringJoiner;
+import java.util.regex.Pattern;
+
+import javax.annotation.Nullable;
+
+final class UrlMapper<T> {
+    private final boolean singlePattern;
+    private final List<Element<T>> elementList = new ArrayList<>();
+
+    UrlMapper(boolean singlePattern) {
+        this.singlePattern = singlePattern;
+    }
+
+    void addMapping(String urlPattern, T object, String objectName) {
+        requireNonNull(urlPattern, "urlPattern");
+        requireNonNull(object, "object");
+        requireNonNull(objectName, "objectName");
+        if (elementList.stream()
+                       .filter(x -> singlePattern && x.name.equals(objectName))
```

Comment:
@dominhhien What is `singlePattern`? Could you explain please?

Reply:
if singlePattern = true, we couldn't override mapping (FilterRegistration). singlePattern = false, we could override (ServletRegistration).

Reply:
Let me just remove it for now because it's always `true`. 😉